TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/kafka/KafkaTrafficCaptureSource.java

```diff
@@ -0,0 +1,221 @@
+        var keepAlivePeriod = getKeepAlivePeriodFromPollPeriod(pollPeriod);
+        return new KafkaTrafficCaptureSource(new KafkaConsumer<>(kafkaProps),
+                topic, keepAlivePeriod, clock, behavioralPolicy);
+    }
+
+    /**
+     * We'll have to 'maintain' touches more frequently than the poll period, otherwise the
+     * consumer will fall out of the group, putting all the commits in-flight at risk.  Notice
+     * that this doesn't have a bearing on heartbeats, which themselves are maintained through
+     * Kafka Consumer poll() calls.  When those poll calls stop, so does the heartbeat, which
+     * is more sensitive, but managed via the 'session.timeout.ms' property.
+     */
+    private static Duration getKeepAlivePeriodFromPollPeriod(Duration pollPeriod) {
+        return pollPeriod.dividedBy(2);
+    }
+
+    public static Properties buildKafkaProperties(@NonNull String brokers,
+                                                  @NonNull String groupId,
+                                                  boolean enableMSKAuth,
+                                                  String propertyFilePath) throws IOException {
+        var kafkaProps = new Properties();
+        kafkaProps.setProperty("key.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
+        kafkaProps.setProperty("value.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer");
+        kafkaProps.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+        kafkaProps.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        if (propertyFilePath != null) {
+            try (InputStream input = new FileInputStream(propertyFilePath)) {
+                kafkaProps.load(input);
+            } catch (IOException ex) {
+                log.error("Unable to load properties from kafka properties file with path: {}", propertyFilePath);
+                throw ex;
+            }
+        }
+        // Required for using SASL auth with MSK public endpoint
+        if (enableMSKAuth) {
+            kafkaProps.setProperty("security.protocol", "SASL_SSL");
+            kafkaProps.setProperty("sasl.mechanism", "AWS_MSK_IAM");
+            kafkaProps.setProperty("sasl.jaas.config", "software.amazon.msk.auth.iam.IAMLoginModule required;");
+            kafkaProps.setProperty("sasl.client.callback.handler.class", "software.amazon.msk.auth.iam.IAMClientCallbackHandler");
+        }
+        kafkaProps.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokers);
+        kafkaProps.setProperty(ConsumerConfig.GROUP_ID_CONFIG, groupId);
+        return kafkaProps;
+    }
+
+    @Override
+    public void touch() {
+        trackingKafkaConsumer.touch();
+    }
+
+    /**
+     * If messages are outstanding, we need to keep the connection alive, otherwise, there's no
+     * reason to.  It's OK to fall out of the group and rejoin once ready.
```

Comment:
Is it? This probably needs to be discussed another time in the context of scalability, but for our single Replayer setup currently is this better than just staying in the group?

Reply:
Is it OK to keep the connection alive when they're outstanding OR to fall out of the group when they are not?
I think both of those points are the best policies (given my current knowledge).  
I'm assuming that "holding" messages for relatively long periods of time is a design trait for the system and that we should allow that to proceed.  That also implies that we'll need other metrics to find out when we're holding messages because of overwork - in which cases we should offload some partitions to other replayers.
The 'leaving the group' part of the statement makes sense to me if the caller REALLY knows that they aren't going to do anything in the interim.  If they change their mind, it's a pretty quick reconnect anyway (albeit with potentially different partition assignments).